### DISCUSSION TITLE
laf product key & roadmap

### DISCUSSION THREAD

**Discussion by maslow** [labring/laf#354] [Announcements]
> 本文介绍和讨论 laf 的产品方向及 Roadmap 相关的问题，欢迎大家在底部参与讨论。

## 🔤 引子
-  解放思想，实事求是
> 做开源产品、做技术产品、做创新产品，做开源技术创新产品，**解放思想**是十分重要的，受经验主义禁锢，就只能制造跑得更快的马车。 同时又不能任由天马行空，沉溺于形而上的幻想，产品不着地，**实事求是**又至关重要。 要做到这两点，既难又易，难在解放思想，易在实事求是。

## ✔️ 产品原则

- 👍 开发者第一
> 以开发者为中心，开发者第一。后面我会写文详细阐述此第一项，如果未来 laf 产品发展方向不以开发者为中心了，应该做一个新产品，不该是 laf。
- 📦 开箱即用
> 开箱即用是 laf 最重要的产品方向和特点
- 🌏 生态建设
> 建设开箱即用的 laf 应用生态，是 laf 的长期大策略
- :octocat: 开源开放
> 开源是基本，开放是视野，关注长期价值，关注广泛价值

### 0️⃣  有用 （2021.Q3 ~ 2022.Q3）
> 现版本 laf v0.8 处于此阶段，数千用户参与使用，有很多社区应用案例，十数家企业用户已用 laf 交付几十个项目，但当前版本使用体验较粗糙、功能不完善、文档滞后，同时存在很多技术债、设计债，早期顶层设计缺乏的弊端逐渐暴露，影响产品的迭代和进程，在进入下一阶段之前，进行了 laf next 的重构工作，此次重构版本目标是发布 v1.0

### 1️⃣  能用 （2022.Q4 ~ 2023.Q1）
> 此阶段的标志是发布 v1.0 版本，代表着 laf 有了比较稳定成熟的 API 和 User Interface 设计实现，并且已准备好用于商业生产环境，公有云可公测付费服务，拥有更完整的功能、文档、demo、产品案例、教程，在进入下一个阶段之前，社区规模会有3～5倍的增长，社区影响力大幅增加，主要开源贡献者扩大 2 倍。

### 2️⃣  好用（2023.Q2 ~ 2023.Q4）
> 此阶段的会在 v1.0 的基础上，打磨用户体验和应用生态。 前者可能体现在多元的 sdk (android/ios)、更友好的UI（web, cli, ide plugin)，后者有代表性的就是「云函数市场」和「应用市场」，丰富应用市场，提高的质量和规模。用户体验不仅体现在良好的开发体验上，还会重点提升终端客户**在应用市场消费应用**的体验。 让开发者通过应用市场与客户连接，帮助开发者更好的分发或销售应用。

### 3️⃣  都用（2024 ~ 2048?）
>  此阶段在目前来看，算是愿景了。 阶段分个一二三，看似是终点，其实没有终点，未来也会不断的重复不同阶段，持续迭代产品。至于产品的终极愿景，我在这里有所提及：
- #178 

## 🏎️  laf 版本计划

| Version       |     Description              | Date  | Status |
| :-: | :- | - | :-: |
| v1.0.0.alpha | 走通核心功能闭环 |  2022.10 | 🏃  |
| v1.0.0.beta  |  完善重要功能，修复绝大部分 bug|  2022.11 |   |
| v1.0.0.rc  | 上线公有云，经过完整系统的测试  |  2022.12 |   |
| v1.0.0|  🦅  |  2023.1 |   |

### ❔  laf 为什么不会有单独的商业版（vs 社区版）?

> laf 所有代码都会是开源的，商业和社区版是100%一致的，不区分，没有理由。

### ❔  laf 怎么盈利？

> 首先，我们融资进展很好，足够支撑产品的研发和迭代。 其次，我们肯定有很完整的商业模式，不然我们也吸引不了投资人，这里就不展开描述（限于篇幅）。

### ❔  laf 何时推出公有云付费服务？

> 有很多用户反馈，希望 laf 能提供公有云付费应用服务，以更放心的将生产应用迁移到公有云上来，也摆脱自运维私有部署版本的烦恼， 限于当前版本并不稳定，且正在进行大版本的迭代，所以计划在 v1.0 发布之后，提供公有云付费服务。

### ❔  laf 何时会发布应用市场？

> v1.0 发布，应用市场会成为优先级最高的一批开发需求。

### ❔  laf 何时提供商业技术支持？

> 当前确实有部分企业客户，希望我们提供付费的技术支持服务，计划在 v1.0 正式发布后，提供标准的商业技术支持服务。

### ❔ 我可以自己私有化部署，运营一个和 lafyun.com 一样的收费公有云平台吗？

> 可以，我希望有更多基于 laf 的公有云平台，做的比 lafyun.com 更好、更大，这是属于 laf 的成功。

### ❔  laf 何时支持 python ?

> 会有计划，要等 v1.0 后，再根据 Roadmap 定优先级。

**Comment by yenche123** [comment]
🚀🚀🚀

**Comment by LeezQ** [comment]
LAF的第一个五年计划 🌝
